packages/basic-crawler/src/internals/basic-crawler.ts

```diff
@@ -150,6 +150,26 @@ export interface BasicCrawlerOptions<
      */
     handleRequestTimeoutSecs?: number;
 
+    /**
+     * A function to handle requests that failed less than `option.maxRequestRetries` times
+     * and would be retried by the crawler by default.
+     * Note that if the function is user-provided, it would not reclaim the request automatically.
+     *
+     * The function receives the following object as an argument:
+     * ```
+     * {
+     *     request: Request,
+     *     error: Error,
```

Comment:
I assume it's only for `errorHandler`? If yes - then done. Or you meant to make 2 args for `failedRequestHandler` too?

Reply:
its a bit weird those two functions have different signatures, we have the error object as part of the failed request context, and here it is in the second arg. 

i would be fine having it in the context and sharing the exact same type as current `failedRequestHandler`, those two should be really the same, the difference is only in when they are executed

Reply:
Oh, but then, wouldn't it make sense to separate the args even in the `failedRequestHandler`. I'm 100% for them having to be the same. But to me it's a bit weird to have to create an extended type for the error handlers. Historically, there was no consistent crawling context, so we've added stuff quite liberally. But now that there is, I wonder if it might be better to do it differently. But I'll leave that up to you.

Reply:
On the other hand, in a fail handler, you will definitely want to access the error - so you might need the second parameter, but not the first one (as you might not care about the context).